packages/visual-editor/src/components/pageSections/EventSection.tsx

```diff
@@ -0,0 +1,231 @@
+              layout="auto"
+              aspectRatio={event.image.image.width / event.image.image.height}
+            />
+          </div>
+        )}
+      </div>
+      <div className="flex flex-col gap-2 p-6 w-full md:w-[55%]">
+        {event.title && <Heading level={6}>{event.title}</Heading>}
+        {event.dateTime?.start && (
+          <Timestamp
+            date={event.dateTime.start}
+            endDate={event.dateTime.end}
+            option={
+              event.dateTime.end
+                ? TimestampOption.DATE_TIME_RANGE
+                : TimestampOption.DATE_TIME
+            }
+            hideTimeZone={true}
+          />
+        )}
+        {event.description?.html && (
+          <div className="font-body-fontFamily font-body-fontWeight text-body-fontSize">
+            <div dangerouslySetInnerHTML={{ __html: event.description.html }} />
+          </div>
+        )}
+        {event.CTA && (
+          <CTA
+            label={event.CTA.label}
+            link={event.CTA.link}
+            linkType={event.CTA.linkType}
+            variant="link"
+          />
+        )}
+      </div>
+    </Background>
+  );
+};
+
+const EventSectionWrapper: React.FC<EventSectionProps> = (props) => {
+  const { styles, sectionHeading, events } = props;
+  const document = useDocument();
+  const resolvedEvents = resolveYextEntityField(document, events);
+  const resolvedHeading = resolveYextEntityField(document, sectionHeading.text);
+
+  return (
+    <PageSection
+      background={styles.backgroundColor}
+      className="flex flex-col gap-8"
+    >
+      {resolvedHeading && (
+        <EntityField
+          displayName="Heading Text"
+          fieldId={sectionHeading.text.field}
+          constantValueEnabled={sectionHeading.text.constantValueEnabled}
+        >
+          <div className="text-center">
+            <Heading level={sectionHeading.level}>{resolvedHeading}</Heading>
+          </div>
+        </EntityField>
+      )}
+      {resolvedEvents && (
+        <EntityField
+          displayName="Events"
+          fieldId={events.field}
+          constantValueEnabled={events.constantValueEnabled}
+        >
+          <div className="flex flex-col gap-8">
+            {resolvedEvents.map((event, index) => (
+              <EventCard
+                key={index}
+                event={event}
+                backgroundColor={styles.cardBackgroundColor}
+              />
+            ))}
+          </div>
+        </EntityField>
+      )}
+    </PageSection>
+  );
+};
+
+export const EventSection: ComponentConfig<EventSectionProps> = {
+  label: "Events Section",
+  fields: eventSectionFields,
+  defaultProps: {
+    styles: {
+      backgroundColor: backgroundColors.background3.value,
+      cardBackgroundColor: backgroundColors.background1.value,
+    },
+    sectionHeading: {
+      text: {
+        field: "",
+        constantValue: "Upcoming Events",
+        constantValueEnabled: true,
```

Comment:
Similar to other PR comment. If there's a default we could either not set it here or on line 219.